packages/backend-test-utils/src/next/services/MockActionsRegistry.ts

```diff
@@ -0,0 +1,164 @@
+{
+  private constructor(private readonly logger: LoggerService) {}
+
+  static create(opts: { logger: LoggerService }) {
+    return new MockActionsRegistry(opts.logger);
+  }
+
+  readonly actions: Map<string, ActionsRegistryActionOptions<any, any>> =
+    new Map();
+
+  async list(): Promise<{ actions: ActionsServiceAction[] }> {
+    return {
+      actions: Array.from(this.actions.entries()).map(([id, action]) => ({
+        id,
+        name: action.name,
+        title: action.title,
+        description: action.description,
+        attributes: {
+          // todo(blam): what's safe defaults?
+          destructive: action.attributes?.destructive ?? false,
+          idempotent: action.attributes?.idempotent ?? false,
+          readOnly: action.attributes?.readOnly ?? false,
+        },
+        schema: {
+          input: action.schema?.input
+            ? zodToJsonSchema(action.schema.input(z))
+            : zodToJsonSchema(z.object({})),
+          output: action.schema?.output
+            ? zodToJsonSchema(action.schema.output(z))
+            : zodToJsonSchema(z.object({})),
+        } as ActionsServiceAction['schema'],
+      })),
+    };
+  }
+
+  async invoke(opts: {
+    id: string;
+    input?: JsonObject;
+    credentials?: BackstageCredentials;
+  }): Promise<{ output: JsonValue }> {
+    const action = this.actions.get(opts.id);
+
+    if (!action) {
+      const availableActionIds = Array.from(this.actions.keys()).join(', ');
+      throw new NotFoundError(
+        `Action "${opts.id}" not found, available actions: ${
+          availableActionIds ? `"${availableActionIds}"` : 'none'
+        }`,
+      );
+    }
+
+    const input = action.schema?.input
+      ? action.schema.input(z).safeParse(opts.input)
+      : ({ success: true, data: undefined } as const);
+
+    if (!input.success) {
+      throw new InputError(`Invalid input to action "${opts.id}"`, input.error);
+    }
+
+    try {
+      const result = await action.action({
+        input: input.data,
+        credentials: opts.credentials ?? mockCredentials.none(),
+        logger: this.logger,
+      });
+
+      const output = action.schema?.output
+        ? action.schema.output(z).safeParse(result?.output)
+        : ({ success: true, data: result?.output } as const);
+
+      if (!output.success) {
+        throw new InputError(
+          `Invalid output from action "${opts.id}"`,
+          output.error,
+        );
+      }
+
+      return { output: output.data };
+    } catch (error) {
+      throw new ForwardedError(
+        `Failed execution of action "${opts.id}"`,
+        error,
+      );
+    }
+  }
+
+  register<
+    TInputSchema extends AnyZodObject,
+    TOutputSchema extends AnyZodObject,
+  >(options: ActionsRegistryActionOptions<TInputSchema, TOutputSchema>): void {
+    // hardcode testing: prefix similar to how the default actions registry does it
+    const id = `testing:${options.name}`;
```

Comment:
I think `test` is best here right? Similar to https://github.com/backstage/backstage/blob/a9219496d5c073aaa0b8caf32ece10455cf65e61/packages/backend-test-utils/src/next/services/mockServices.ts#L321

and

https://github.com/backstage/backstage/blob/861f162b4a39117b824669d67a951ed1db142e3d/packages/backend-test-utils/src/next/wiring/ServiceFactoryTester.ts#L99